grant-project/zcash-grant-system
# RFP Fields (matching, bounty, dates)

Closes #84.

### What This Does

* Adds `matching`, `bounty`, `date_closes`, `date_opened`, and `date_closed` to the RFP model.
* Adds admin fields for setting matching, bounty, and date_closes.
* Displays the above 3 on RFPs
* Proposals submitted against RFPs with matching automatically set `contribution_matching = 1.0`
* Proposals made from RFPs no longer take the title and brief
* RFPs will have their status soft-set to `CLOSED` after `date_closes`. This won't be reflected in the db, but will show up everywhere else. Potentially shift this to a task so that it's updated in the db.

### Steps to Test

* Update your db, confirm that your old RFPs were given `date_opened` and `date_closed` fields correctly.
* Create a new RFP, confirm that the new fields save and display properly.
* Create a new proposal off of the RFP, confirm that only the category is inherited and the `contribution_matching` was correctly set.
* Create an RFP with the `date_closes` set in the past, confirm it shows as `CLOSED`.

## Screenshots

### List display with closes date

<img width="821" alt="screen shot 2019-02-08 at 1 53 06 pm" src="https://user-images.githubusercontent.com/649992/52500711-a7237980-2bac-11e9-8b56-8b77bc02f3e0.png">

<br/>
<br/>

#### Detail display showing new fields at the bottom

<img width="874" alt="screen shot 2019-02-08 at 2 04 12 pm" src="https://user-images.githubusercontent.com/649992/52500721-ae4a8780-2bac-11e9-8a6d-40fde017bdbe.png">

<br/>
<br/>

#### New admin fields

<img width="919" alt="screen shot 2019-02-08 at 2 17 51 pm" src="https://user-images.githubusercontent.com/649992/52500732-b276a500-2bac-11e9-9813-fee75e34a28b.png">

<br/>
<br/>

#### Status field when date_closes is in the past

<img width="423" alt="screen shot 2019-02-08 at 2 18 13 pm" src="https://user-images.githubusercontent.com/649992/52500755-bdc9d080-2bac-11e9-89b9-734418cc02b4.png">
